### PR TITLE
Add a merge script.

### PR DESCRIPTION
**Pull request by sellout**
Should currently work with any Scala repo. Would be nice to generalize
for others – basically comes down to handling different version files.

### CODE REVIEW

**Comment by sellout** [comment]
Uhh … there’s no travis file. Don’t know why it tried to build.

File: bin/merge [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    git merge --no-ff --no-edit $tmp_branch
+    old_version=($(sed 's/.*"\(.*\)"/\1/' version.sbt | tr "." "\n"))
+    echo $old_version
+    if [[ -e "CHANGELOG/major.md" ]]; then
+        new_version="${old_version[0]+1}.0.0"
+        mv CHANGELOG/major.md CHANGELOG/${new_version}.md
+    elif [[ -e "CHANGELOG/minor.md" ]]; then
+        new_version="${old_version[0]}.${old_version[1]+1}.0"
+        mv CHANGELOG/minor.md CHANGELOG/${new_version}.md
+    else
```
Comment (mossprescott) [inline bin/merge]: Better to be pedantic lest we get it wrong if someone misspells `minr` or `mjaor`?

Comment (mossprescott) [inline bin/merge]: Also, wouldn't we want to fail if `revision.md` is missing?

File: bin/merge [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        new_version="${old_version[0]}.${old_version[1]}.${old_version[2]+1}"
+        mv CHANGELOG/revision.md CHANGELOG/${new_version}.md
```
Comment (mossprescott) [inline bin/merge]: Maybe that's already going to happen here, in which case I guess that's fine.

**Comment by mossprescott** [comment]
I mistyped the repo name and got a confusing error:

```
> merge quasar-analytisc/quasar 1110
curl: (22) The requested URL returned error: 404 Not Found
error: quasar-analytisc/quasar#1110 has already been merged
```

**Comment by mossprescott** [comment]
Playing with JR's current PR, a couple of things I notice:

There are two merge commits in the history. Both seem to come from the script (and have me as the author):

```
15ea04e Merge branch 'jedesah-topic/upgrade_sbt'
cd0183e Merge branch 'topic/upgrade_sbt' of https://github.com/jedesah/Quasar into jedesah-topic/upgrade_sbt
620c5f2 Upgrade to sbt 0.13.11
```

Also, since he didn't have a `revision.md`, there's no version bump :(

**Comment by jedesah** [comment]
To be fair though. That PR is just bumping the sbt version so I don't think it warrants a quasar version bump.

**Comment by mossprescott** [comment]
I also notice that `echo $old_version` prints `4` and `new_version` seems to be 4.1.1 (should be 4.1.3).

**Comment by jedesah** [comment]
But the script should probably error out or emit a warning in that case.

**Comment by mossprescott** [comment]
@jedesah: it looks like the version didn't get bumped because the script doesn't work, not because of anything you did, not that we have a way to say "no need to release with this commit".

File: bin/merge (hunk 1 above)
Comment (sellout) [inline bin/merge]: Yeah, good point.

File: bin/merge (hunk 2 above)
Comment (sellout) [inline bin/merge]: It won’t, because of the lack of `-e`, but I should `unset -e`/`set -e` around the one command, rather than shutting it off completely.

**Comment by sellout** [comment]
I hate that you can’t reply to specific comments …
1. typoed-repo-name: Yeah, that error is because of the lack of -e, otherwise, you’d just get the “requested URL returned error” message (which, c’mon curl, include the URL!)
2. a) I think there are two merge commits because his branch isn’t rebased. I tried that PR earlier, but it didn’t work because of merge conflicts, which is good. b) Hrmm, I was getting increments without a `revision.md` … I don’t think not having that file should prevent an increment (although it will fail the script once I make the change mentioned above)
3. Every change to the repo warrants a version bump.
4. a) old-version is an array – you’re only seeing the first value when you echo it. b) ah-ha! The incrementing must actually just be setting to “1”. I was testing on a matryoshka PR, so it was going from 0→1, but 3→1 is clearly not correct.

**Comment by jedesah** [comment]
@sellout I am not sure I agree about a change to the Readme, bumping the
version of quasar.

**Comment by sellout** [comment]
README is documentation, docs are versioned.

**Comment by jedesah** [comment]
Depends on what kind of doc. Scaladoc is packaged in the jar so it makes
sense to bump the artifact version. However the Readme is not published in
the jar, that's not to say it is not versioned, but it's versioned by git,
not our artifact version.

**Comment by sellout** [comment]
Ping, re: new commits.

**Comment by mossprescott** [comment]
LGTM

**Pull request merged**